src/cps/utils.cpp

```diff
@@ -19,4 +21,14 @@ namespace cps::utils {
         return out;
     }
 
+    std::string_view trim(std::string_view input) {
+        input.remove_prefix(std::distance(input.cbegin(), std::find_if(input.cbegin(), input.cend(), [](const int ch) {
+                                              return !std::isspace(ch);
+                                          })));
+        input.remove_suffix(
+            std::distance(input.crbegin(), std::find_if(input.crbegin(), input.crend(),
+                                                        [](const int ch) { return !std::isspace(ch); })));
```

Comment:
Do you mind breaking these up maybe six statements connected by meaningful local variable names? I expect clarity would improve that way.